### PR TITLE
Feature: add ability to pass in specific version of svelte

### LINKED ISSUE #123: Support for multiple Svelte versions?
I've been working with a very sizable business app that's been stuck in Svelte 1 land for... a while.

Because it's reasonably large, transitioning the whole site in one go would be difficult, but it's all in one project so I'm trying to work out ways to transition chunks to Svelte 3.

With npm v6.9.0 you can alias different versions of the same library, so you could do:

```json
"dependencies": {
   "svelte1": "npm:svelte@1",
   "svelte3": "npm:svelte@3"
}
```

However, this plugin puts the Svelte version checking at initialization, e.g. in [index.js](https://github.com/sveltejs/rollup-plugin-svelte/blob/master/index.js) it has:

```js
const { version } = require('svelte/package.json');
// ...
const major_version = +version[0];
// ...
const { compile, preprocess } = major_version >= 3
	? require('svelte/compiler.js')
	: require('svelte');
```

## Proposal?

Imagine if the version of Svelte could be passed in as an option:

```js
import svelte1 from 'svelte1' // through the magic of alias
import svelte3 from 'svelte3'
import sveltePlugin from 'rollup-plugin-svelte'

export default {
  // ...
  plugins: [
    sveltePlugin({
      svelte: svelte1,
      extensions: [ '.html' ]
    }),
    sveltePlugin({
      svelte: svelte3,
      extensions: [ '.svelte' ]
    }),
    // ...
```

It looks like the `compile` and `preprocess` are only used in the returned `transform` function, so this could be a non-breaking change:

```js
function loadSvelte() {
	const { version } = require('svelte/package.json');
	const major_version = +version[0];
	return major_version >= 3
		? require('svelte/compiler.js')
		: require('svelte');
}
module.exports = function svelte(options = {}) {
	const { compile, preprocess } = options.svelte || loadSvelte()
```

I think the general idea is sound, but improvements could probably be made.

### PR DESCRIPTION
**Pull request by saibotsivad**
This will resolve #123 by adding support for passing in a specific version of the Svelte compiler (more discussion on that issue).

My skills at TypeScript are sorely lacking, so I'm not confident that I added definitions correctly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,6 +77,16 @@ export default {
...
+      },
+
+      // Pass in a specific version of Svelte, e.g. if you use
+      // npm aliases to install multiple versions you can import
+      // a specific version:
+      // import svelte1 from 'svelte1' // aliased version of svelte@1.x
+      svelte: {
+        compile: svelte1.compile,
+        preprocess: svelte1.preprocess,
+        major_version: 1
```
Comment (antony) [inline README.md:89]: all of the other options within the plugin are camelCased, so I think this should be too.

File: index.d.ts [hunk 2]
```diff
@@ -23,6 +23,12 @@ declare class CssWriter {
   toString(): string;
 }
 
+interface Svelte {
+  compile: any;
+  preprocess: any;
+  major_version: number;
```
Comment (antony) [inline index.d.ts:29]: I think this could just be renamed to `version` and perhaps typed to be an integer, if that's a thing.

**Review by antony — CHANGES_REQUESTED** [review]
Syntactical change

**Comment by lukeed** [comment]
Made the requested changes. Also made it so that the `Svelte` interface, which is now using `version`, is `require`-friendly. You can pass in individual keys, as was originally intended, or use `require('svelte')` directly. I updated the README to reflect this. @saibotsivad If you don't mind taking a peek at this again, would be great 🙇

**Comment by lukeed** [comment]
OK. We have decided that this will be backported as a `6.1.0` release... and then removed in the upcoming 7.0 release. The reasoning is that start with 7.0, we're dropping support for old Node/Rollup/Svelte versions. We have to do this for sanity, and semver allows us to do so (see #138)

Once I get the [`6.x`](https://github.com/sveltejs/rollup-plugin-svelte/tree/6.x) branch up to date, I will update this base branch & revert the rebase that I applied to this PR.

**Review by benmccann — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
 const { encode, decode } = require('sourcemap-codec');
 
-const major_version = +version[0];
 const pkg_export_errors = new Set();
 
-const { compile, preprocess } = major_version >= 3
-	? require('svelte/compiler.js')
-	: require('svelte');
+// future: breaks on v10.0
+const toMajor = str => Number(str[0]);
```
Comment (Conduitry) [inline index.js:10]: Can we snake case this to `to_major`? And the limitation of breaking at v10 seems pretty remote at this point, but it seems easy to address, with a `str.split('.')[0]` or a `str.match(/^[^.]+/)[0]` or whatever.

Comment (lukeed) [reply index.js:10]: The beauty of `6.x` is that we'll now never get there 😆 But ya I can adjust

**Pull request merged**